### DISCUSSION TITLE
There are problems with frequency-depence complex permitivity for THz radiation

### DISCUSSION THREAD

**Discussion by leSemaleon** [NanoComp/meep#2679] [Q&A]
Terahertz radiation is long-wavelength, so the structures I want to study have dimensions of the order of 1000 microns. This creates resolution restrictions. I use a resolution of about 0.05 pixels per micron in my simulations. I read that for example copper, even 10 pixels per micron is too low a resolution, as an error appears about fields is NAN or infinity. How to solve the situation?

**Comment by smartalecH** [comment]
>  I read that for example copper, even 10 pixels per micron is too low a resolution, as an error appears about fields is NAN or infinity

In what context? If you're just reading old issues and discussions here, keep in mind that 99% of the applications here are in the optical domain. So resolution heuristics are on an entirely different wavelength scale (plus the material definitions for eg copper are completely different). 

Set up your simulation and run a simple convergence check. Note you don't have to use a micron as your fundamental unit length. Rather, I would choose something closer to your wavelength (maybe choose 100 um). Then, be sure your material definitions are appropriately referenced to this scale. Finally, slowly ramp up the resolution until you see convergence.
